pkg/collector/worker/utilization_tracker.go

```diff
@@ -117,9 +117,7 @@ func newUtilizationTrackerWithClock(
 		return utilization
 	}
 
-	ut.statsUpdateFunc = func(sw util.SlidingWindow) {
-		utilization := sw.Average()
-
+	ut.statsUpdateFunc = func(utilization float64) {
```

Comment:
Not a blocker and would be a bigger change but I wonder if `statsUpdateFunc` is a good name anymore that we always only return the average.